internal/controller/controllers/agent_controller.go

```diff
@@ -516,6 +516,46 @@ func (r *AgentReconciler) shouldApproveCSRsForAgent(ctx context.Context, agent *
 	return false, nil
 }
 
+func marshalNodeLabels(nodeLabels map[string]string) (string, error) {
+	b, err := json.Marshal(&nodeLabels)
+	if err != nil {
+		return "", err
+	}
+	return string(b), err
+}
+
+func (r *AgentReconciler) applyDay2NodeLabels(ctx context.Context, log logrus.FieldLogger, agent *aiv1beta1.Agent, client spoke_k8s_client.SpokeK8sClient) error {
+	if agent.Status.Progress.CurrentStage == models.HostStageDone && !funk.IsEmpty(agent.Spec.NodeLabels) {
```

Comment:
Why we want to wait for done? It will not be too late?

Reply:
https://github.com/openshift/assisted-service/pull/5042/files#diff-ec0c8ee92a41e93de3f6906e7ddd408c175553ef5d57b247e7d0b56ec7a7be47L641 
we are setting Done in case node is Ready so or we should allow setting labels before or we should not check NodeIsReady here i assume. Though i think better to allow setting it before as in day2 setting Done actually means we finished the installation and i think we should not finish it till we did all the operations. IMO we should set labels and Done should be set only after we finish all node operations

Reply:
Removed check on stage.  Will only check node